imblearn/utils/fixes.py

```diff
@@ -0,0 +1,26 @@
+"""Compatibility fixes for older version of python, numpy and scipy
+If you add content to this file, please give the version of the package
+at which the fix is no longer needed.
+
+Backdated from scikit-learn.
+"""
+# Authors: Emmanuelle Gouillart <emmanuelle.gouillart@normalesup.org>
+#          Gael Varoquaux <gael.varoquaux@normalesup.org>
+#          Fabian Pedregosa <fpedregosa@acm.org>
+#          Lars Buitinck
+#
+# License: BSD 3 clause
+
+from sklearn.externals._packaging.version import parse as parse_version
+import scipy
+import scipy.stats
+
+
+sp_version = parse_version(scipy.__version__)
+
+
+# TODO: Remove when SciPy 1.9 is the minimum supported version
+def _mode(a, axis=0):
+    if sp_version >= parse_version("1.9.0"):
+        return scipy.stats.mode(a, axis=axis, keepdims=True)
+    return scipy.stats.mode(a, axis=axis)
```

Comment:
I found `sklearn.utils.fixes._mode` in `scikit-learn==1.1.2`:

```bash
pip install scikit-learn==1.1.2
python -c 'import sklearn; print(sklearn.utils.fixes._mode)'
```

We could add the checks for `scikit-learn>=1.1.2` like follows (but I think the current fix is simpler and basically does the same thing):

```python
from sklearn.externals._packaging.version import parse as parse_version
import scipy
import scipy.stats
import sklearn

sk_version = parse_version(sklearn.__version__)
sp_version = parse_version(scipy.__version__)


def _mode(a, axis=0):
    if sp_version >= parse_version("1.9.0"):
        # TODO: Remove when SciPy 1.9 is the minimum supported version
        return scipy.stats.mode(a, axis=axis, keepdims=True)
    elif sk_version >= parse_version("1.1.2"):
        # TODO: Remove when scikit-learn 1.2 is the minimum supported version
        return sklearn.utils.fixes._mode(a, axis=axis)
    return scipy.stats.mode(a, axis=axis)
```